metrics/resource.go

```diff
@@ -21,21 +21,24 @@ func getNodeResource(s stats.NodeStats, metadata *Metadata) *Resource {
 		LabelNodeName: s.NodeName,
 	}
 
+	resource := &Resource{
+		Type:   "node",
+		Name:   s.NodeName,
+		Labels: labels,
+	}
+	if s.CPU != nil {
+		resource.Timestamp = s.CPU.Time.Time
```

Comment:
Do we know what happens when we don't set the timestamp?

Reply:
Based on our conversation yesterday, looks like it sets the timestamp on the event based on the resource timestamp [`ev.Timestamp = res.Timestamp`](https://github.com/honeycombio/honeycomb-kubernetes-agent/blob/c062f64c5db08c9668459f81fc089c2fa494d230/service/runnable.go#L177)  using libhoney in `runnable.go`. In that case there's likely a timestamp set on ingest.

Is it better to have an else clause here so a timestamp still exists, setting the Timestamp equal to the start time on the pod, node, etc?

```go
	if s.CPU != nil {
		resource.Timestamp = s.CPU.Time.Time
	} else {
		resource.Timestamp = s.StartTime.Time
	}
```

Reply:
Oh actually, looks like `Timestamp` is optional to set the time for the event, and otherwise [defaults to `time.Now()`](https://github.com/honeycombio/libhoney-go/blob/v1.15.8/libhoney.go#L891). So either way this should be okay to omit the timestamp unless we think it's better to have something.

For some additional testing I also curled the events endpoint with a true Timestamp `"X-Honeycomb-Event-Time: 2022-01-07T20:47:26.794Z"` and an empty Timestamp `"X-Honeycomb-Event-Time: "` and the empty time used server ingest time.

Reply:
I wonder if @puckpuck has an opinion here, as the OG author 😉